### ISSUE TITLE
multi_gpu_model renames model outputs

### ISSUE THREAD

**Issue by anatolix** [keras-team/keras#8512]
This code in multi_gpu_model actually renames output to "concatenate_1", "concatenate_2" ... etc

https://github.com/fchollet/keras/blob/master/keras/utils/training_utils.py
```
 with tf.device('/cpu:0'):
        merged = []
        for outputs in all_outputs:
            merged.append(concatenate(outputs,
                                      axis=0))
        return Model(model.inputs, merged)
```

This cause problems while assigning loss function with dict and understanding metrics on tensorboard.
Need set same names for concatenate as underlying model output names
